pkg/piecestore/rpc/server/server.go

```diff
@@ -0,0 +1,172 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package server
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	"golang.org/x/net/context"
+
+	"storj.io/storj/pkg/piecestore"
+	"storj.io/storj/pkg/piecestore/rpc/server/ttl"
+	pb "storj.io/storj/protos/piecestore"
+)
+
+// Server -- GRPC server meta data used in route calls
+type Server struct {
+	PieceStoreDir string
+	DB            *ttl.TTL
+}
+
+// Store -- Store incoming data using piecestore
+func (s *Server) Store(stream pb.PieceStoreRoutes_StoreServer) error {
+	log.Println("Storing data...")
+
+	total := int64(0)
+
+	// Receive initial meta data about what's being stored
+	piece, err := stream.Recv()
+	if err != nil {
+		return err
+	}
+
+	if err := s.DB.AddTTLToDB(piece.Id, piece.Ttl); err != nil {
+		return err
+	}
+
+	// Initialize file for storing data
+	storeFile, err := pstore.StoreWriter(piece.Id, piece.Size, piece.StoreOffset, s.PieceStoreDir)
+	if err != nil {
+		return err
+	}
+	defer storeFile.Close()
+
+	for {
+		storeMsg, err := stream.Recv()
```

Comment:
if you turn `stream` into a io.Reader (perhaps making PieceStreamReader more general), then you can just do io.Copy here too.

Reply:
👍